hyperspy/_signals/spectrum.py

```diff
@@ -84,7 +84,10 @@ def _spikes_diagnosis(self, signal_mask=None,
             dc = dc[~navigation_mask, :]
         der = np.abs(np.diff(dc, 1, -1))
         plt.figure()
-        plt.hist(np.ravel(der.max(-1)), 100)
+        if len(dc.shape) == 1:
```

Comment:
Good. Actually, taking the max is there in order to decrease the burden on the `hist` function when using the tool on a large number of spectra. However, it's probably not needed (and actually counterproductive) when the number of spectra (n) is below 1000. Here you fix the corner case n =1, but one can argue that this is also broken (although it doesn't raise an exception) for n < 100. So what about:

```python

n = (~navigation_mask).sum() if navigation_mask else self.axes_manager.navigation_size
if n < 1000:
    plt.hist(der, 100)
else:
    plt.hist(np.ravel(der.max(-1)), 100)

```

Also, now that we're at this, it might be worth to replace matplotlib's `hist` with hyperspy's `get_histogram` which actually uses astroML code to automatically determine the number of bins. By default it uses the freedman rule (which is probably a good choice for the purpose), but there is also the option of using non-uniform bin sizes.